suite-common/wallet-core/src/fiat-rates/fiatRatesMiddleware.ts

```diff
@@ -1,72 +1,23 @@
-import { selectAccounts } from '../accounts/accountsReducer';
-import {
-    removeFiatRatesForDisabledNetworksThunk,
-    updateCurrentFiatRatesThunk,
-    initFiatRatesThunk,
-    updateLastWeekFiatRatesThunk,
-    updateTxsFiatRatesThunk,
-    onUpdateFiatRateThunk,
-} from './fiatRatesThunks';
+import { fetchFiatRatesThunk } from './fiatRatesReducer';
 import { blockchainActions } from '../blockchain/blockchainActions';
+import { updateTxsFiatRatesThunk } from './fiatRatesThunks';
 
 export const prepareFiatRatesMiddleware = createMiddlewareWithExtraDeps(
-    (action, { dispatch, extra, next, getState }) => {
+    (action, { dispatch, extra, next }) => {
         const {
-            actions: { setWalletSettingsLocalCurrency, changeWalletSettingsNetworks },
+            actions: { setWalletSettingsLocalCurrency },
         } = extra;
-        const prevStateAccounts = selectAccounts(getState());
-
-        if (accountsActions.updateAccount.match(action)) {
-            // fetch rates for new tokens added on account update
-            const account = action.payload;
-            const prevAccount = prevStateAccounts.find(
-                a => a.descriptor === account.descriptor && a.symbol === account.symbol,
-            );
-
-            if (account.tokens) {
-                const difference = account.tokens.filter(
-                    token =>
-                        !prevAccount?.tokens?.find(prevToken => prevToken.symbol === token.symbol),
-                );
 
-                difference.forEach(token => {
-                    if (token.symbol) {
-                        dispatch(
-                            updateCurrentFiatRatesThunk({
-                                ticker: {
-                                    symbol: token.symbol,
-                                    mainNetworkSymbol: account.symbol,
-                                    tokenAddress: token.address,
-                                },
-                            }),
-                        );
-                    }
-                });
-            }
-        }
-
-        if (accountsActions.createAccount.match(action)) {
-            // fetch current rates for account's tokens
-            const account = action.payload;
-            account.tokens?.forEach(token => {
-                if (!token.symbol) {
-                    return;
-                }
-                dispatch(
-                    updateCurrentFiatRatesThunk({
-                        ticker: {
-                            symbol: token.symbol,
-                            mainNetworkSymbol: account.symbol,
-                            tokenAddress: token.address,
-                        },
-                    }),
-                );
-            });
+        if (isAnyOf(accountsActions.updateAccount, accountsActions.createAccount)) {
+            dispatch(fetchFiatRatesThunk({ rateType: 'current' }));
+            dispatch(fetchFiatRatesThunk({ rateType: 'lastWeek' }));
```

Comment:
this is an infinite loop, I guess? or somewhere else inside this middleware

Reply:
Why do you think? Update of fiat rates should not trigger any of these actions nor transactions actions.